fiftyone/utils/ultralytics.py

```diff
@@ -270,3 +289,107 @@ def _to_detections(self, df):
                 for row in df.itertuples()
             ]
         )
+
+
+class FiftyOneYOLOModel(Model):
+    """YOLO model wrapper for FiftyOne.
+
+    Args:
+        config: a dict which has the `ultralytics.YOLO` model as the value for
+            the key `model`
+    """
+
+    def __init__(self, config):
+        self.model = config.get("model", None)
+
+    @property
+    def media_type(self):
+        return "image"
+
+    @property
+    def ragged_batches(self):
+        return True
+
+    @property
+    def transforms(self):
+        return None
+
+    @property
+    def preprocess(self):
+        return False
+
+    def predict(self, args):
+        raise NotImplementedError(
+            "Subclass must implement `predict` or `predict_all`"
```

Comment:
nit: remove `predict_all` from this error message. Subclasses **have** to implement `predict`